crates/net/network/src/swarm.rs

```diff
@@ -199,9 +200,16 @@ where
                         }
                         InboundConnectionError::ExceedsLimit(limit) => {
                             // We currently don't have additional capacity to establish a session
-                            // from an incoming connection, so we drop
-                            // the connection.
-                            trace!(target: "net", %limit, ?remote_addr, "Exceeded incoming connection limit; dropping connection");
+                            // from an incoming connection, so we send a disconnect message with
+                            // TooManyPeers reason.
+                            trace!(target: "net", %limit, ?remote_addr, "Exceeded incoming connection limit; disconnecting");
+
+                            tokio::task::spawn(async move {
```

Comment:
this needs to be moved to Session Manager and also spawned with the configured executor if it's set.